graph/src/loader.c

```diff
@@ -33,27 +39,12 @@ void rage_element_loader_free(rage_ElementLoader * el) {
     free(el);
 }
 
-static int is_plugin(const struct dirent * entry) {
-    struct stat s;
-    // TODO: remove this pathname match tat once in separate packages
-    if (strncmp(entry->d_name, "lib", 3) != 0) {
-        return false;
-    }
-    if (strncmp(entry->d_name, "librage", 7) == 0) {
-        return false;
-    }
-    if (stat(entry->d_name, &s) == 0) {
-        return S_ISREG(s.st_mode);
-    }
-    return false;
-}
-
 rage_ElementTypes * rage_element_loader_list(rage_ElementLoader * el) {
     rage_ElementTypes * elems = malloc(sizeof(rage_ElementTypes));
     elems->len = 0;
     elems->items = NULL;
     struct dirent ** entries;
-    int n_items = scandir(el->elems_path, &entries, is_plugin, NULL);
+    int n_items = scandir(el->elems_path, &entries, el->pf, NULL);
```

Comment:
I presume `scandir` is happy if you provide it with a `NULL` pointer?

Reply:
Yup: http://man7.org/linux/man-pages/man3/scandir.3.html
"If filter is NULL all entries are selected"